doc/setup-commands.rst

```diff
@@ -1034,6 +1034,27 @@ Miscellaneous options
     Specify a soft constraint on versions of a package. The solver will
     attempt to satisfy these preferences on a "best-effort" basis.
 
+.. option:: --enable-buildinfo
+
+    Generate accurate build information for build components.
+
+    Information contain meta information, such as component type, compiler type, and Cabal library version used to build,
+    but also fine grained information, such as dependencies, what modules are part of the component, etc...
+
+    On build, a file ``build-info.json`` (in the ``json`` format) will be written to the root of the build directory.
+
+    .. note::
+        The format and fields of the generated build information is currently experimental,
+        in the future we might add or remove fields, depending on the needs of other tooling.
+
+    .. jsonschema:: ./json-schemas/build-info.schema.json
```

Comment:
Having the schema sure is useful but I think it would still be good to have a (perhaps abbreviated) example json here.

Reply:
Added an example!